src/utils/websockets.js

```diff
@@ -38,11 +40,21 @@ export function connectWS(topic, onMessageCallback) {
   };
 
   const connect = () => {
+    let reconnectCounter = 0;
     this.sockClient = new Client({
       brokerURL: config.WS_URL,
-      // debug: function(str) {
-      //   console.log('debug: ', str);
-      // },
+      debug(str) {
+        // console.log('debug: ', str);
+        // -- detect reconnect and increment the reconnect counter
+        if (str.includes('reconnect')) {
+          reconnectCounter += 1;
+        }
```

Comment:
what happening in following case:
 we have like 3 failures and then one successful connect.
After a while we have another failure.

Is the counter restarted after each successful connect?

Reply:
If I were to reset the counter I would be in the same situation and will be a continuous loop.

Reply:
As to this particular PR i fully agree with @teosarca : AFAIU it's not usable
As to possible solutions to the actual problem: I *think* the web-socket client should have some sort of lifecycle management with methods like https://stomp-js.github.io/api-docs/latest/classes/Client.html#onWebSocketError
or *some* way of detecting the API retuning `401`.
Maybe you can consult with @siemiatj if you need a more concrete starting point?

Reply:
@metas-ts @teosarca I've checked with both onWebSocketError and also with onStompError (which already exists in the code) and nighter catches anything. I'm open to other solutions @siemiatj

Reply:
@metas-ts  @petrican  @siemiatj  when developing/suggesting a solution pls keep in mind the initial problem. i.e. if we stop the server and start it again, frontend tries to connect using the old session (cookie) hence we have this flood.

Reply:
> As to this particular PR i fully agree with @teosarca : AFAIU it's not usable
> As to possible solutions to the actual problem: I _think_ the web-socket client should have some sort of lifecycle management with methods like https://stomp-js.github.io/api-docs/latest/classes/Client.html#onWebSocketError
> or _some_ way of detecting the API returning `401`.
> Maybe you can consult with @siemiatj if you need a more concrete starting point?

So there are two issues:

1)  network goes down and websocket goes down also. User need to be shown an Offline mode when that happens
For that case I could not identify a proper way to know when network is down. You don't get anything from BE as status because it never reaches BE. Tried two ways
   a) hook on debug message and looking for 'reconnect' string - tested it works you can also tune number of max retries
   b) hook on errorCallback - would be same thing but will be called once only without the posibility to tune retries. Also it returns on sigle string like -> "Whoops! Lost connection to http://localhost:18080/stomp"

2) Network goes down and then connectivity is up again. Here I managed to detect 401 and redirect to root. https://github.com/metasfresh/metasfresh-webui-frontend/pull/2629/files#diff-537f7fefa17eccb1e4d6e78dbf8b2b2dR221 User is going to input credentials and there will be a new session. Any subsequent ws connection will take place without issues. When the 401 is detected by redirecting you also stop the flood of failed attempts.

Reply:
I vote to go with `b)` in first case. If there's no connection I don't really see a reason to retry hitting the backend. Users will refresh the window anyway.

Reply:
added as extra safety the check for user session